packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -534,7 +534,7 @@ export class Repository extends RepositoryBase {
       // It says "Text", but they actually mean "Object".
       repositoryPolicyText: Lazy.any({ produce: () => this.policyDocument }),
       lifecyclePolicy: Lazy.any({ produce: () => this.renderLifecyclePolicy() }),
-      imageScanningConfiguration: props.imageScanOnPush ? { scanOnPush: true } : { scanOnPush: false },
+      imageScanningConfiguration: props.imageScanOnPush ? { scanOnPush: props.imageScanOnPush } : undefined,
```

Comment:
My earlier suggestion was wrong (and syntactically incorrect doh!). It should be

```js
imageScanningConfiguration: props.imageScanOnPush === undefined ? undefined : { scanOnPush: props.imageScanOnPush }
```

Reply:
You're right, I'll fix that up and add a test to verify functionality